### PR TITLE
Clear out with/close/try/finally

### PR DESCRIPTION
**Pull request by kshyatt**
Replaced some `close` calls with explicit `free` on pointers.

I did a little bit of whitespace prettying as well.

### CODE REVIEW

File: base/libgit2/signature.jl [hunk 1]
```diff
@@ -15,15 +15,13 @@ function Signature(name::AbstractString, email::AbstractString)
     @check ccall((:git_signature_now, :libgit2), Cint,
                  (Ptr{Ptr{SignatureStruct}}, Cstring, Cstring), sig_ptr_ptr, name, email)
     sig = GitSignature(sig_ptr_ptr[])
-    s = Signature(sig.ptr)
-    close(sig)
+    s   = Signature(sig.ptr)
```
Comment (simonbyrne) [inline base/libgit2/signature.jl]: `return Signature(sig.ptr)` ?

Comment (kshyatt) [inline base/libgit2/signature.jl]: Good catch!

**Comment by simonbyrne** [comment]
We probably should wait until we've fixed the segfaults (#20109/#20124).

**Comment by kshyatt** [comment]
Fine with me. I'm happy to continue to do style cleanup on this PR while we wait.

File: base/libgit2/libgit2.jl [hunk 2]
```diff
@@ -174,26 +158,27 @@ function branch!(repo::GitRepo, branch_name::AbstractString,
...
         try
             new_branch_ref = create_branch(repo, branch_name, cmt, force=force)
-        finally
-            if new_branch_ref === nothing
-                throw(GitError(Error.Object, Error.ERROR, "cannot create branch `$branch_name` with `$commit_id`"))
+        catch err
+            if isa(err, GitError)
+                throw(GitError(Error.Object, err.Code, "cannot create branch `$branch_name` with `$commit_id`"))
+            else
+                rethrow(e)
```
Comment (tkelman) [inline base/libgit2/libgit2.jl]: err

Comment (kshyatt) [inline base/libgit2/libgit2.jl]: Derp, thanks <3

**Comment by simonbyrne** [comment]
Looks like some files under `pkg` still use `with`: these need to be removed as well.

**Comment by kshyatt** [comment]
Is the arrayops thing to do with me?

**Comment by kshyatt** [comment]
@Keno @tkelman is the Windows EPERM fail here my fault?

**Comment by tkelman** [comment]
Likely. That happens if you try to delete a file that you still have open handles to.

**Comment by simonbyrne** [comment]
It seems to be failing on `Pkg.rm` (though different places in each Windows build).

**Comment by kshyatt** [comment]
This is hella old and conflicted so I'm going to close (ironic, I know) and try again.

**Pull request closed without merging**